akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -359,127 +368,116 @@ object ShardCoordinator {
  * @see [[ClusterSharding$ ClusterSharding extension]]
  */
 class ShardCoordinator(typeName: String, settings: ClusterShardingSettings,
-                       allocationStrategy: ShardCoordinator.ShardAllocationStrategy)
-  extends PersistentActor with ActorLogging {
+                       allocationStrategy: ShardCoordinator.ShardAllocationStrategy,
+                       replicator: ActorRef)
+  extends Actor with ActorLogging with Stash {
   import ShardCoordinator._
   import ShardCoordinator.Internal._
   import ShardRegion.ShardId
   import settings.tuningParameters._
 
-  override def persistenceId = s"/sharding/${typeName}Coordinator"
-
-  override def journalPluginId: String = settings.journalPluginId
-
-  override def snapshotPluginId: String = settings.snapshotPluginId
-
   val removalMargin = Cluster(context.system).settings.DownRemovalMargin
+  val waitingForStateTimeout = settings.tuningParameters.waitingForStateTimeout
+  val updatingStateTimeout = settings.tuningParameters.updatingStateTimeout
 
-  var persistentState = State.empty
+  var state = State.empty
   var rebalanceInProgress = Set.empty[ShardId]
   var unAckedHostShards = Map.empty[ShardId, Cancellable]
   // regions that have requested handoff, for graceful shutdown
   var gracefulShutdownInProgress = Set.empty[ActorRef]
   var aliveRegions = Set.empty[ActorRef]
-  var persistCount = 0
 
   import context.dispatcher
   val rebalanceTask = context.system.scheduler.schedule(rebalanceInterval, rebalanceInterval, self, RebalanceTick)
 
   Cluster(context.system).subscribe(self, ClusterShuttingDown.getClass)
 
+  implicit val node = Cluster(context.system)
+  val CoordinatorStateKey = LWWRegisterKey[State](s"${typeName}CoordinatorState")
```

Comment:
Later we could try to implement the member age aware register, using the `akka.cluster.Member.isOlderThan`, we want to use the data from the newest singleton. That can be rather useful for singletons in general.

If you want to take a stab at that new data type you find useful info here: http://doc.akka.io/docs/akka/snapshot/scala/distributed-data.html#Custom_Data_Type
and the following commit is a good example of what pieces that are required: 7041c76ba910e4474b1004060c5cbc08a121b5e1